file/file.go

```diff
@@ -142,20 +142,13 @@ func (s *fileSub) Close() error {
 
 // ----------------------------------------------------------------------------
 
-type subTopic struct {
-	filename string
-}
-
-// NewSubTopic inits a subscriber topic for given file name.
+// SubTopic is an adapter for filename, that behaves like bps.SubTopic.
 // Useful for testing.
-func NewSubTopic(filename string) bps.SubTopic {
-	return &subTopic{
-		filename: filename,
-	}
-}
+type SubTopic string
```

Comment:
wait. does it need to be public?

Reply:
I think so - it's much easier to test if it's public if you use `bps.SubTopic` in your code. Just give it a dummy file.